app/models/manageiq/providers/embedded_ansible/crud_common.rb

```diff
@@ -0,0 +1,111 @@
+module ManageIQ::Providers::EmbeddedAnsible::CrudCommon
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def queue(zone, instance_id, method_name, args, action, auth_user)
+      task_opts = {
+        :action => action,
+        :userid => auth_user || "system"
+      }
+
+      queue_opts = {
+        :args        => args,
+        :class_name  => name,
+        :method_name => method_name,
+        :role        => "ems_operations", # TODO: This should go to a git_owner
```

Comment:
What is this actually doing? It probably doesn't really need a role if it's just doing the object CRUD as it's not actually talking to a provider anymore, right?

Reply:
@carbonin well, wouldn't this be determining which worker is able to pick up a job, and since this is a shared lib, that could include both running the playbooks and cloning them.

For right now, I think ideally we only want to be doing that on the appliances that have the embedded ansible role, correct?  (I could be thinking about this wrong as well)

Reply:
I think the only thing that will go through this method boils down to the `raw_create_in_provider` invocation which is just creating AR objects.

@Fryguy can you weigh in here?

Reply:
I think for now, this should be on the embedded ansible role.  Part of the issue is that previously *all* actions went to Tower, so they needed a Task + Notification.  So, for now, we still need that even though it's as simple as a DB create.  On later refactoring and cleanup though, I agree with @carbonin that we can be more selective about what goes where, perhaps not even using the queue at all.

Reply:
> I think the only thing that will go through this method boils down to the raw_create_in_provider invocation which is just creating AR objects.

I'm not sure, because I didn't get into the other objects like credentials, but you may be right.

Reply:
Yeah, the "for now" was the big question being asked here.  But yeah, I will make the change then.

* * *

> I think the only thing that will go through this method boils down to the raw_create_in_provider invocation which is just creating AR objects.

Just for some From `EmbeddedAnsible::ConfigurationScriptSource`:

```ruby
  def self.raw_create_in_provider(manager, params)
    params.delete(:scm_type)   if params[:scm_type].blank?
    params.delete(:scm_branch) if params[:scm_branch].blank?

    transaction { create!(params.merge(:manager => manager)).tap(&:sync) }
end
```

(`sync` being what does the git operations)

Which, _**I think**_ based on the previous implementation where tower was involved, this is still done through the queue because we would have to wait to hear back from tower that everything was cloned properly.